python/test/unit/language/test_core.py

```diff
@@ -37,6 +39,19 @@ def is_hip():
     THREADS_PER_WARP = 32
 
 
+def add_interpreter_test(func):
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        func(*args, **kwargs)  # run regular
+
+        os.environ['TRITON_INTERPRET'] = '1'
+        func(*args, **kwargs)  # run interpreter mode
+        os.environ['TRITON_INTERPRET'] = '0'
+
+    return wrapper
```

Comment:
- If `regular` throws an exception, this won't run `interpreter`, which seems undesirable.
- If `interpreter` throws an exception, this will not reset the envvar, which is definitely undesirable.
- The two runs of the test will be reported as one test by pytest, so if one fails, you won't be able to tell whether it's the interpreter or the regular one which failed.

I think this needs more pytest magic to make the two tests separate tests as far as pytest is concerned.